### PR TITLE
Added N-Queens, Count-Leaf-Nodes & Deleted factorial

### PR DESCRIPTION
**Pull request by adisakshya**
Let me know if it's OK.

### CODE REVIEW

File: pygorithm/Backtracking/N-Queen.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# Python program to solve N Queen 
+# Problem using backtracking 
+
+global N 
+N = 4
+
+def printSolution(board):
```
Comment (OmkarPathak) [inline pygorithm/Backtracking/N-Queen.py]: Change function name to `print_solution` so as follow Python PEP8 styling

File: pygorithm/Backtracking/N-Queen.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+		print
+
+
+# A utility function to check if a queen can 
+# be placed on board[row][col]. Note that this 
+# function is called when "col" queens are 
+# already placed in columns from 0 to col -1. 
+# So we need to check only left side for 
+# attacking queens 
+def isSafe(board, row, col):
```
Comment (OmkarPathak) [inline pygorithm/Backtracking/N-Queen.py]: Change function name to `is_safe` so as follow Python PEP8 styling

File: pygorithm/Backtracking/N-Queen.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+			return False
+
+	# Check lower diagonal on left side 
+	for i,j in zip(range(row,N,1), range(col,-1,-1)): 
+		if board[i][j] == 1: 
+			return False
+
+	return True
+
+def solveNQUtil(board, col):
```
Comment (OmkarPathak) [inline pygorithm/Backtracking/N-Queen.py]: Change function name to `solve_nq_util` so as follow Python PEP8 styling

File: pygorithm/Backtracking/N-Queen.py [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+# This function solves the N Queen problem using 
+# Backtracking. It mainly uses solveNQUtil() to 
+# solve the problem. It returns false if queens 
+# cannot be placed, otherwise return true and 
+# placement of queens in the form of 1s. 
+# note that there may be more than one 
+# solutions, this function prints one of the 
+# feasible solutions. 
+def solveNQ():
```
Comment (OmkarPathak) [inline pygorithm/Backtracking/N-Queen.py]: Change function name to `solve_nq` so as follow Python PEP8 styling

File: pygorithm/data_structures/tree.py [hunk 5]
```diff
@@ -405,6 +405,17 @@ def number_of_nodes(self, root):
 
         return left_number + right_number + 1
     
+    def getLeafCount(node):
```
Comment (OmkarPathak) [inline pygorithm/data_structures/tree.py]: Change function name to `get_leaf_count` so as follow Python PEP8 styling

File: pygorithm/math/factorial.py [hunk 6]
```diff
@@ -1,34 +1,10 @@
 """
```
Comment (OmkarPathak) [inline pygorithm/math/factorial.py]: Delete this file, as there is a reason why I didn't use builtin factorial from`math` module. Main purpose behind this module is to teach people how to implement various DS and Algos in Python. Hence, we don't need this file as this is already implemented

File: pygorithm/Backtracking/N-Queen.py (hunk 1 above)
Comment (adisakshya) [reply pygorithm/Backtracking/N-Queen.py]: Done

File: pygorithm/Backtracking/N-Queen.py (hunk 2 above)
Comment (adisakshya) [reply pygorithm/Backtracking/N-Queen.py]: Done

File: pygorithm/Backtracking/N-Queen.py (hunk 3 above)
Comment (adisakshya) [reply pygorithm/Backtracking/N-Queen.py]: Done

File: pygorithm/Backtracking/N-Queen.py (hunk 4 above)
Comment (adisakshya) [reply pygorithm/Backtracking/N-Queen.py]: Done

File: pygorithm/data_structures/tree.py (hunk 5 above)
Comment (adisakshya) [reply pygorithm/data_structures/tree.py]: Done

File: pygorithm/math/factorial.py (hunk 6 above)
Comment (adisakshya) [reply pygorithm/math/factorial.py]: Done

**Pull request closed without merging**